src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -153,24 +165,108 @@ void GcsPlacementGroupManager::HandleCreatePlacementGroup(
+  const auto placement_group_id =
+      PlacementGroupID::FromBinary(request.placement_group_id());
+
+  RemovePlacementGroup(placement_group_id, [send_reply_callback, reply,
+                                            placement_group_id](Status status) {
+    if (status.ok()) {
+      RAY_LOG(INFO) << "Placement group of an id, " << placement_group_id
+                    << " is removed successfully.";
+    }
+    GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
+  });
+}
+
+void GcsPlacementGroupManager::RemovePlacementGroup(
+    const PlacementGroupID &placement_group_id,
+    StatusCallback on_placement_group_removed) {
+  RAY_CHECK(on_placement_group_removed);
+  // If the placement group has been already removed, don't do anything.
+  auto placement_group_it = registered_placement_groups_.find(placement_group_id);
+  if (placement_group_it == registered_placement_groups_.end()) {
+    on_placement_group_removed(Status::OK());
+    return;
+  }
+  auto placement_group = placement_group_it->second;
+  registered_placement_groups_.erase(placement_group_it);
+
+  // Destroy all bundles.
+  gcs_placement_group_scheduler_->DestroyPlacementGroupBundleResourcesIfExists(
+      placement_group_id);
+  // Cancel the scheduling request if necessary.
+  if (IsSchedulingInProgress(placement_group_id)) {
+    // If the placement group is scheduling.
+    gcs_placement_group_scheduler_->MarkScheduleCancelled(placement_group_id);
+  }
+
+  // Remove a placement group from a pending list if exists.
+  auto pending_it = std::find_if(
+      pending_placement_groups_.begin(), pending_placement_groups_.end(),
+      [placement_group_id](const std::shared_ptr<GcsPlacementGroup> &placement_group) {
+        return placement_group->GetPlacementGroupID() == placement_group_id;
+      });
+  if (pending_it != pending_placement_groups_.end()) {
+    // The placement group was pending scheduling, remove it from the queue.
+    pending_placement_groups_.erase(pending_it);
+  }
+
+  // Flush the status and respond to workers.
+  placement_group->UpdateState(rpc::PlacementGroupTableData::REMOVED);
+  RAY_CHECK_OK(gcs_table_storage_->PlacementGroupTable().Put(
+      placement_group->GetPlacementGroupID(),
+      placement_group->GetPlacementGroupTableData(),
+      [this, on_placement_group_removed, placement_group_id](Status status) {
+        RAY_CHECK_OK(status);
+        // If placement group hasn't been created yet, send a response to a core worker
+        // that the creation of placement group has failed.
+        auto it = placement_group_to_register_callback_.find(placement_group_id);
+        if (it != placement_group_to_register_callback_.end()) {
+          std::stringstream stream;
+          stream << "Placement group of id " << placement_group_id
+                 << " is removed before it is created";
+          it->second(Status::NotFound(stream.str()));
```

Comment:
Not sure it's needed to put the group id in the string, presumably the caller already knows it and can print out a message including that already.

Reply:
Agree. It'll be also printed in the core worker side code. I will just remove the id string,